### ISSUE TITLE
crashy aot

### ISSUE THREAD

**Issue by timsgardner** [arcadia-unity/Arcadia#201]
git bisect indicates this was introduced at 95f9e70a75e93341974a635edf61c3589fb0ff3f, the commit that upgraded to Clojure 1.8.

This aot has long thrown an error at some point in its process, yet worked, in that we end up with the expected DLLs in Arcadia/Compiled:
```clojure
(arcadia.compiler/aot-namespaces
  "Assets/Arcadia/Compiled"
  '[clojure.core
    arcadia.core
    arcadia.repl
    arcadia.config
    arcadia.introspection
    arcadia.literals
    arcadia.packages
    arcadia.compiler])
```
As of 95f9e70a75e93341974a635edf61c3589fb0ff3f, Unity crashes as soon as it tries to load the generated DLLs, and crashes immediately on all subsequent attempts to open it until Library and Arcadia/Compiled are deleted.

This AOT seems to work:
```clojure
(arcadia.compiler/aot-namespaces
  "Assets/Arcadia/Compiled"
  '[arcadia.core, arcadia.literals, arcadia.internal.name-utils, arcadia.internal.messages])
```

**Comment by timsgardner** [comment]
This seems to be related to 1.8 trying to set private or readonly properties on vars when they are redefined in a new namespace. This happens in arcadia.internal.functions. 

This can be temporarily resolved by just not redefining vars (via refer-clojure :exclude, for example). Not a great fix though, better solution is to make the problematic fields on vars non-readonly or (better?) keep the compiler from attempting to mutate them.

**Comment by nasser** [comment]
I will work on a compiler patch for this

**Comment by timsgardner** [comment]
presumed closed as of a6b8fa7c2a23a731d2d2b01a66b1276ccdc984bc (probably before)

**Comment by nasser** [comment]
Strictly speaking, the compiler was never patched to prevent this, but this is no longer a problem for us. Not sure if this will rear its head down the line again.
